src/middlewares/authentication.ts

```
@@ -32,80 +32,215 @@
-        'route' in def &&
-        def.route.path === ctx.req.path &&
-        def.route.method.toUpperCase() === ctx.req.method,
-    );
+    let matchedPath = ctx.req.path;
+    let basePath = '';
+
+    // Try to use matchedRoutes for better route matching
+    // Guard against undefined matchedRoutes (some Hono versions may not set it)
+    // Prefer the last match (most specific/deepest) over parent routes by reversing
+    const normalizedMethod = ctx.req.method.toUpperCase();
+    let matchedRoute: any = undefined;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)